src/components/menu/UserItems.tsx

```diff
@@ -98,18 +100,20 @@ export const UserItems: FC<IUserItemsProps> = ({
 					<GLink size='Big'>{formatMessage({ id: i.title })}</GLink>
 				</Item>
 			))}
-			<Item
-				onClick={() => {
-					isSignedIn && dispatch(signOut(token!));
-					localStorage.removeItem(StorageLabel.WALLET);
-					disconnect();
-				}}
-				theme={theme}
-			>
-				<GLink size='Big'>
-					{formatMessage({ id: 'label.sign_out' })}
-				</GLink>
-			</Item>
+			{!isSafeEnv && (
+				<Item
+					onClick={() => {
```

Comment:
Could you please use useCallback to cache it during re-renders. Ref: https://react.dev/reference/react/useCallback#usecallback

Reply:
I don't think a useCallback is necessary, but in words of chatGPT advice

`Using useCallback isn't necessary in this case because it's meant for memoizing functions to prevent unnecessary re-renders when passing functions to child components. Since useIsSafeEnvironment is concerned with updating state in response to changing connectors, useEffect is already the appropriate hook for the job. Adding useCallback would not offer any performance benefits and would only complicate the code without any clear advantage.`

https://github.com/Giveth/giveth-dapps-v2/blob/e10a2f13994c489fa6c04d657103e7b53900e39d/src/hooks/useSafeAutoConnect.tsx#L25-L34

let me know if we still wanna do the change